From: Gunnar Wrobel <p@rdus.de>
Subject: [PATCH] t/GLOBAL/HK/GW/ConfigTemplateSupport

This adds a small section to each configuration file that allows
overriding the default settings with a local configuration file.
These local overrides are going to be generated via templates for
the Kolab Server.

https://www.intevation.de/roundup/kolab/issue3420

Notes:

Did this + a few manual edits ->

 for FILE in `find horde-webmail/ -wholename *config/*.php | grep -v scripts`;do A=${FILE/*config\//}; B=${A/.php/.local.php};echo "if (file_exists(dirname(__FILE__) . '/$B')) {
  require_once(dirname(__FILE__) . '/$B');
}
" >> $FILE;done

for FILE in `find horde-webmail/ -wholename *config/*.php | grep -v scripts`;do A=${FILE/*config\//}; B=${A/.php/.local.php};C=${FILE//\//_};D=${C/horde-webmail_/};E=${D/_config/};echo "KOLAB_META_START
TARGET=@@@horde_confdir@@@/$B
PERMISSIONS=0400
OWNERSHIP=@@@webserver_usr@@@:@@@webserver_grp@@@
KOLAB_META_END
<?php
/* This file can be used to customize your kolab-webclient installation. */
/* Anything you add to this configuration file will override settings    */
/* from @@@horde_confdir@@@/$A
   Please refer to that file to see which settings you can influence via */
/* this template.                                                        */" > /CVS/server/kolab-webclient/webclient-${E};done


Signed-off-by: Gunnar Wrobel <p@rdus.de>

---
 horde-webmail/config/hooks.php              |    4 ++++
 horde-webmail/config/mime_drivers.php       |    4 ++++
 horde-webmail/config/motd.php               |    7 +++++++
 horde-webmail/config/nls.php                |    4 ++++
 horde-webmail/config/prefs.php              |    4 ++++
 horde-webmail/config/registry.php           |    4 ++++
 horde-webmail/dimp/config/conf.php          |    4 ++++
 horde-webmail/dimp/config/hooks.php         |    4 ++++
 horde-webmail/dimp/config/menu.php          |    4 ++++
 horde-webmail/dimp/config/portal.php        |    4 ++++
 horde-webmail/dimp/config/prefs.php         |    4 ++++
 horde-webmail/dimp/config/servers.php       |    4 ++++
 horde-webmail/imp/config/conf.php           |    4 ++++
 horde-webmail/imp/config/header.php         |    4 ++++
 horde-webmail/imp/config/hooks.php          |    4 ++++
 horde-webmail/imp/config/menu.php           |    4 ++++
 horde-webmail/imp/config/mime_drivers.php   |    4 ++++
 horde-webmail/imp/config/motd.php           |    4 ++++
 horde-webmail/imp/config/prefs.php          |    4 ++++
 horde-webmail/imp/config/servers.php        |    4 ++++
 horde-webmail/imp/config/spelling.php       |    4 ++++
 horde-webmail/ingo/config/backends.php      |    4 ++++
 horde-webmail/ingo/config/conf.php          |    4 ++++
 horde-webmail/ingo/config/fields.php        |    4 ++++
 horde-webmail/ingo/config/hooks.php         |    4 ++++
 horde-webmail/ingo/config/prefs.php         |    4 ++++
 horde-webmail/kronolith/config/keywords.php |    4 ++++
 horde-webmail/kronolith/config/menu.php     |    4 ++++
 horde-webmail/kronolith/config/prefs.php    |    4 ++++
 horde-webmail/mimp/config/conf.php          |    4 ++++
 horde-webmail/mimp/config/menu.php          |    4 ++++
 horde-webmail/mimp/config/mime_drivers.php  |    4 ++++
 horde-webmail/mimp/config/prefs.php         |    4 ++++
 horde-webmail/mnemo/config/conf.php         |    4 ++++
 horde-webmail/mnemo/config/prefs.php        |    4 ++++
 horde-webmail/nag/config/conf.php           |    4 ++++
 horde-webmail/nag/config/menu.php           |    4 ++++
 horde-webmail/nag/config/prefs.php          |    4 ++++
 horde-webmail/turba/config/attributes.php   |    4 ++++
 horde-webmail/turba/config/conf.php         |    4 ++++
 horde-webmail/turba/config/hooks.php        |    4 ++++
 horde-webmail/turba/config/menu.php         |    4 ++++
 horde-webmail/turba/config/mime_drivers.php |    4 ++++
 horde-webmail/turba/config/prefs.php        |    4 ++++
 horde-webmail/turba/config/sources.php      |    4 ++++
 45 files changed, 183 insertions(+), 0 deletions(-)

diff --git a/horde-webmail/config/hooks.php b/horde-webmail/config/hooks.php
index 1fb2f6e..c7e41a2 100644
--- a/horde-webmail/config/hooks.php
+++ b/horde-webmail/config/hooks.php
@@ -691,3 +691,7 @@ if (!function_exists('_prefs_hook_fullname')) {
         }
     }
 }*/
+if (file_exists(dirname(__FILE__) . '/hooks.local.php')) {
+  require_once(dirname(__FILE__) . '/hooks.local.php');
+}
+
diff --git a/horde-webmail/config/mime_drivers.php b/horde-webmail/config/mime_drivers.php
index e963402..da0e14a 100644
--- a/horde-webmail/config/mime_drivers.php
+++ b/horde-webmail/config/mime_drivers.php
@@ -638,3 +638,7 @@ $mime_drivers['horde']['wordperfect'] = array(
         'default' => 'wordperfect.png'
     )
 );
+if (file_exists(dirname(__FILE__) . '/mime_drivers.local.php')) {
+  require_once(dirname(__FILE__) . '/mime_drivers.local.php');
+}
+
diff --git a/horde-webmail/config/motd.php b/horde-webmail/config/motd.php
index 9f0b57b..bca4c41 100644
--- a/horde-webmail/config/motd.php
+++ b/horde-webmail/config/motd.php
@@ -1,6 +1,13 @@
 <!-- This file contains any "Message Of The Day" Type information -->
 <!-- It will be included below the log-in form on the login page. -->
 
+<?php
+if (file_exists(dirname(__FILE__) . '/motd.local.php')) {
+  require_once(dirname(__FILE__) . '/motd.local.php');
+}
+?>
+
 <br />
 <table width="100%"><tr><td align="center"><img src="themes/graphics/horde-power1.png" alt="Powered by Horde" /></td></tr></table>
 
+
diff --git a/horde-webmail/config/nls.php b/horde-webmail/config/nls.php
index 21a3d19..8720096 100644
--- a/horde-webmail/config/nls.php
+++ b/horde-webmail/config/nls.php
@@ -646,3 +646,7 @@ $tz['Pacific/Wallis'] = 'Pacific/Wallis';
 $tz['Pacific/Yap'] = 'Pacific/Yap';
 
 $GLOBALS['tz'] = &$tz;
+if (file_exists(dirname(__FILE__) . '/nls.local.php')) {
+  require_once(dirname(__FILE__) . '/nls.local.php');
+}
+
diff --git a/horde-webmail/config/prefs.php b/horde-webmail/config/prefs.php
index 52c7730..d3632dc 100644
--- a/horde-webmail/config/prefs.php
+++ b/horde-webmail/config/prefs.php
@@ -574,3 +574,7 @@ $_prefs['editor_plugins'] = array(
     'desc' => _("Select editor plugins")
 );
 // End Rich Text Editor preferences.
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/config/registry.php b/horde-webmail/config/registry.php
index cbd6221..6268106 100644
--- a/horde-webmail/config/registry.php
+++ b/horde-webmail/config/registry.php
@@ -525,3 +525,7 @@ function _detect_webroot()
 
     return $webroot;
 }
+if (file_exists(dirname(__FILE__) . '/registry.local.php')) {
+  require_once(dirname(__FILE__) . '/registry.local.php');
+}
+
diff --git a/horde-webmail/dimp/config/conf.php b/horde-webmail/dimp/config/conf.php
index de4e32c..83a50d4 100644
--- a/horde-webmail/dimp/config/conf.php
+++ b/horde-webmail/dimp/config/conf.php
@@ -15,3 +15,7 @@ $conf['viewport']['background_inbox'] = true;
 $conf['search']['search_all'] = false;
 $conf['menu']['apps'] = array('turba', 'ingo', 'kronolith', 'nag', 'mnemo');
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/dimp/config/hooks.php b/horde-webmail/dimp/config/hooks.php
index f315631..02a6dc7 100644
--- a/horde-webmail/dimp/config/hooks.php
+++ b/horde-webmail/dimp/config/hooks.php
@@ -123,3 +123,7 @@
 //         return $ret;
 //     }
 // }
+if (file_exists(dirname(__FILE__) . '/hooks.local.php')) {
+  require_once(dirname(__FILE__) . '/hooks.local.php');
+}
+
diff --git a/horde-webmail/dimp/config/menu.php b/horde-webmail/dimp/config/menu.php
index 5dcf354..842c4c5 100644
--- a/horde-webmail/dimp/config/menu.php
+++ b/horde-webmail/dimp/config/menu.php
@@ -28,3 +28,7 @@
 //         'text' => 'Say Hi!',
 //         'icon' => $GLOBALS['registry']->getImageDir('horde') . '/horde.png'),
 // );
+if (file_exists(dirname(__FILE__) . '/menu.local.php')) {
+  require_once(dirname(__FILE__) . '/menu.local.php');
+}
+
diff --git a/horde-webmail/dimp/config/portal.php b/horde-webmail/dimp/config/portal.php
index 7720eb3..3ccf2ae 100644
--- a/horde-webmail/dimp/config/portal.php
+++ b/horde-webmail/dimp/config/portal.php
@@ -47,3 +47,7 @@ $dimp_block_list[$collection->getName('turba', 'minisearch')] = array(
 $dimp_block_list[$collection->getName('kronolith', 'summary')] = array(
     'ob' => $collection->getBlock('kronolith', 'summary', array())
 );
+if (file_exists(dirname(__FILE__) . '/portal.local.php')) {
+  require_once(dirname(__FILE__) . '/portal.local.php');
+}
+
diff --git a/horde-webmail/dimp/config/prefs.php b/horde-webmail/dimp/config/prefs.php
index a55ecc2..09a0041 100644
--- a/horde-webmail/dimp/config/prefs.php
+++ b/horde-webmail/dimp/config/prefs.php
@@ -30,3 +30,7 @@ $_prefs['show_preview'] = array(
     'shared' => false,
     'type' => 'implicit',
 );
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/dimp/config/servers.php b/horde-webmail/dimp/config/servers.php
index 30962f3..b0c5de9 100644
--- a/horde-webmail/dimp/config/servers.php
+++ b/horde-webmail/dimp/config/servers.php
@@ -253,3 +253,7 @@ if ($GLOBALS['conf']['kolab']['enabled']) {
         ),
     );
 }
+if (file_exists(dirname(__FILE__) . '/servers.local.php')) {
+  require_once(dirname(__FILE__) . '/servers.local.php');
+}
+
diff --git a/horde-webmail/imp/config/conf.php b/horde-webmail/imp/config/conf.php
index 3a0f1d6..011472d 100644
--- a/horde-webmail/imp/config/conf.php
+++ b/horde-webmail/imp/config/conf.php
@@ -63,3 +63,7 @@ $conf['sentmail']['driver'] = 'none';
 $conf['tasklist']['use_tasklist'] = true;
 $conf['notepad']['use_notepad'] = true;
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/imp/config/header.php b/horde-webmail/imp/config/header.php
index a442e09..7cc941e 100644
--- a/horde-webmail/imp/config/header.php
+++ b/horde-webmail/imp/config/header.php
@@ -20,3 +20,7 @@
 $_header = array();
 
 /* Add your custom entries below this line. */
+if (file_exists(dirname(__FILE__) . '/header.local.php')) {
+  require_once(dirname(__FILE__) . '/header.local.php');
+}
+
diff --git a/horde-webmail/imp/config/hooks.php b/horde-webmail/imp/config/hooks.php
index 3bab14e..3c39f20 100644
--- a/horde-webmail/imp/config/hooks.php
+++ b/horde-webmail/imp/config/hooks.php
@@ -535,3 +535,7 @@ if (!function_exists('_imp_hook_quota')) {
         return array($quota[1] * 1024, $quota[2] * 1024);
     }
 }
+if (file_exists(dirname(__FILE__) . '/hooks.local.php')) {
+  require_once(dirname(__FILE__) . '/hooks.local.php');
+}
+
diff --git a/horde-webmail/imp/config/menu.php b/horde-webmail/imp/config/menu.php
index 1ea56c7..0d2925c 100644
--- a/horde-webmail/imp/config/menu.php
+++ b/horde-webmail/imp/config/menu.php
@@ -39,3 +39,7 @@
 $_menu = array();
 
 /* Add your custom entries below this line. */
+if (file_exists(dirname(__FILE__) . '/menu.local.php')) {
+  require_once(dirname(__FILE__) . '/menu.local.php');
+}
+
diff --git a/horde-webmail/imp/config/mime_drivers.php b/horde-webmail/imp/config/mime_drivers.php
index 1019aaf..786934b 100644
--- a/horde-webmail/imp/config/mime_drivers.php
+++ b/horde-webmail/imp/config/mime_drivers.php
@@ -301,3 +301,7 @@ $mime_drivers['imp']['tnef'] = array(
         'default' => 'binary.png'
     )
 );
+if (file_exists(dirname(__FILE__) . '/mime_drivers.local.php')) {
+  require_once(dirname(__FILE__) . '/mime_drivers.local.php');
+}
+
diff --git a/horde-webmail/imp/config/motd.php b/horde-webmail/imp/config/motd.php
index 70803c9..60a33f5 100644
--- a/horde-webmail/imp/config/motd.php
+++ b/horde-webmail/imp/config/motd.php
@@ -36,6 +36,10 @@
 //
 // echo '<a class="small" href="' . $SERVER_HTTP_URL . '" target="_parent">' . _("Click here for Standard HTTP") . '</a> - <a class="small" href="' . $SERVER_SSL_URL . '" target="_parent">' . _("Click here for Secure HTTPS") . '</a></div>';
 
+if (file_exists(dirname(__FILE__) . '/motd.local.php')) {
+  require_once(dirname(__FILE__) . '/motd.local.php');
+}
 ?>
 <br />
 <table width="100%"><tr><td align="center"><?php echo Horde::img('horde-power1.png', _("Powered by Horde"), '', $registry->getImageDir('horde')) ?></td></tr></table>
+
diff --git a/horde-webmail/imp/config/prefs.php b/horde-webmail/imp/config/prefs.php
index e1f6426..02a28ab 100644
--- a/horde-webmail/imp/config/prefs.php
+++ b/horde-webmail/imp/config/prefs.php
@@ -1506,3 +1506,7 @@ $_prefs['vinbox_id'] = array(
     'locked' => false,
     'shared' => false,
     'type' => 'implicit');
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/imp/config/servers.php b/horde-webmail/imp/config/servers.php
index 1c6e537..5fd0cf7 100644
--- a/horde-webmail/imp/config/servers.php
+++ b/horde-webmail/imp/config/servers.php
@@ -286,3 +286,7 @@ if ($GLOBALS['conf']['kolab']['enabled']) {
         ),
     );
 }
+if (file_exists(dirname(__FILE__) . '/servers.local.php')) {
+  require_once(dirname(__FILE__) . '/servers.local.php');
+}
+
diff --git a/horde-webmail/imp/config/spelling.php b/horde-webmail/imp/config/spelling.php
index e9b78bd..7bbe2c2 100644
--- a/horde-webmail/imp/config/spelling.php
+++ b/horde-webmail/imp/config/spelling.php
@@ -11,3 +11,7 @@ $ignore_list = array(
     'jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct',
     'nov', 'dec', 'fwd', 'http', 'https', 'html', 'email', 'bcc', 'jpg', 'gif'
 );
+if (file_exists(dirname(__FILE__) . '/spelling.local.php')) {
+  require_once(dirname(__FILE__) . '/spelling.local.php');
+}
+
diff --git a/horde-webmail/ingo/config/backends.php b/horde-webmail/ingo/config/backends.php
index d5e26c1..feeabb6 100644
--- a/horde-webmail/ingo/config/backends.php
+++ b/horde-webmail/ingo/config/backends.php
@@ -327,3 +327,7 @@ if ($GLOBALS['conf']['kolab']['enabled']) {
         'shares' => false
     );
 }
+if (file_exists(dirname(__FILE__) . '/backends.local.php')) {
+  require_once(dirname(__FILE__) . '/backends.local.php');
+}
+
diff --git a/horde-webmail/ingo/config/conf.php b/horde-webmail/ingo/config/conf.php
index ffc834a..8a35d6f 100644
--- a/horde-webmail/ingo/config/conf.php
+++ b/horde-webmail/ingo/config/conf.php
@@ -13,3 +13,7 @@ $conf['spam']['compare'] = 'string';
 $conf['hooks']['vacation_addresses'] = false;
 $conf['hooks']['vacation_only'] = true;
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/ingo/config/fields.php b/horde-webmail/ingo/config/fields.php
index 045ff52..82fc932 100644
--- a/horde-webmail/ingo/config/fields.php
+++ b/horde-webmail/ingo/config/fields.php
@@ -104,3 +104,7 @@ $ingo_fields = array(
                      'matches', 'not matches')
   )
 );
+if (file_exists(dirname(__FILE__) . '/fields.local.php')) {
+  require_once(dirname(__FILE__) . '/fields.local.php');
+}
+
diff --git a/horde-webmail/ingo/config/hooks.php b/horde-webmail/ingo/config/hooks.php
index 9e52c95..9e9685c 100644
--- a/horde-webmail/ingo/config/hooks.php
+++ b/horde-webmail/ingo/config/hooks.php
@@ -22,3 +22,7 @@
 //         return array($user . '@example.com', $user . '@foobar.com');
 //     }
 // }
+if (file_exists(dirname(__FILE__) . '/hooks.local.php')) {
+  require_once(dirname(__FILE__) . '/hooks.local.php');
+}
+
diff --git a/horde-webmail/ingo/config/prefs.php b/horde-webmail/ingo/config/prefs.php
index fe645fb..0f7641e 100644
--- a/horde-webmail/ingo/config/prefs.php
+++ b/horde-webmail/ingo/config/prefs.php
@@ -112,3 +112,7 @@ $_prefs['spam'] = array(
     'shared' => false,
     'type' => 'implicit'
 );
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/kronolith/config/keywords.php b/horde-webmail/kronolith/config/keywords.php
index 6ed32d5..f227f74 100644
--- a/horde-webmail/kronolith/config/keywords.php
+++ b/horde-webmail/kronolith/config/keywords.php
@@ -15,3 +15,7 @@
  */
 
 $keywords[_("Type")] = array(_("Birthday"), _("Meeting"), _("Game"));
+if (file_exists(dirname(__FILE__) . '/keywords.local.php')) {
+  require_once(dirname(__FILE__) . '/keywords.local.php');
+}
+
diff --git a/horde-webmail/kronolith/config/menu.php b/horde-webmail/kronolith/config/menu.php
index 282ac8a..f097cb6 100644
--- a/horde-webmail/kronolith/config/menu.php
+++ b/horde-webmail/kronolith/config/menu.php
@@ -39,3 +39,7 @@
 $_menu = array();
 
 /* Add your custom entries below this line. */
+if (file_exists(dirname(__FILE__) . '/menu.local.php')) {
+  require_once(dirname(__FILE__) . '/menu.local.php');
+}
+
diff --git a/horde-webmail/kronolith/config/prefs.php b/horde-webmail/kronolith/config/prefs.php
index 22c08c3..099c385 100644
--- a/horde-webmail/kronolith/config/prefs.php
+++ b/horde-webmail/kronolith/config/prefs.php
@@ -497,3 +497,7 @@ $_prefs['last_kronolith_maintenance'] = array(
     'shared' => false,
     'type' => 'implicit'
 );
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/mimp/config/conf.php b/horde-webmail/mimp/config/conf.php
index 29762a5..dcd6165 100644
--- a/horde-webmail/mimp/config/conf.php
+++ b/horde-webmail/mimp/config/conf.php
@@ -4,3 +4,7 @@
 $conf['mailbox']['max_from_chars'] = 10;
 $conf['mailbox']['max_subj_chars'] = 20;
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/mimp/config/menu.php b/horde-webmail/mimp/config/menu.php
index 2f0a5c7..46bbf9e 100644
--- a/horde-webmail/mimp/config/menu.php
+++ b/horde-webmail/mimp/config/menu.php
@@ -27,3 +27,7 @@
 $_menu = array();
 
 /* Add your custom entries below this line. */
+if (file_exists(dirname(__FILE__) . '/menu.local.php')) {
+  require_once(dirname(__FILE__) . '/menu.local.php');
+}
+
diff --git a/horde-webmail/mimp/config/mime_drivers.php b/horde-webmail/mimp/config/mime_drivers.php
index 1666a86..6c0843c 100644
--- a/horde-webmail/mimp/config/mime_drivers.php
+++ b/horde-webmail/mimp/config/mime_drivers.php
@@ -100,3 +100,7 @@ $mime_drivers['mimp']['multipart'] = array(
         'multipart/*'
     )
 );
+if (file_exists(dirname(__FILE__) . '/mime_drivers.local.php')) {
+  require_once(dirname(__FILE__) . '/mime_drivers.local.php');
+}
+
diff --git a/horde-webmail/mimp/config/prefs.php b/horde-webmail/mimp/config/prefs.php
index 52bbb95..1d3ddf6 100644
--- a/horde-webmail/mimp/config/prefs.php
+++ b/horde-webmail/mimp/config/prefs.php
@@ -23,3 +23,7 @@ $_prefs['preview_msg'] = array(
     'desc' => _("Display only the first 250 characters of a message initially?"));
 
 // End Message Viewing preferences
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/mnemo/config/conf.php b/horde-webmail/mnemo/config/conf.php
index a4e9764..864f011 100644
--- a/horde-webmail/mnemo/config/conf.php
+++ b/horde-webmail/mnemo/config/conf.php
@@ -7,3 +7,7 @@ $conf['menu']['print'] = true;
 $conf['menu']['import_export'] = true;
 $conf['menu']['apps'] = array();
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/mnemo/config/prefs.php b/horde-webmail/mnemo/config/prefs.php
index b83d5a4..09133c7 100644
--- a/horde-webmail/mnemo/config/prefs.php
+++ b/horde-webmail/mnemo/config/prefs.php
@@ -116,3 +116,7 @@ $_prefs['delete_opt'] = array(
     'type' => 'checkbox',
     'desc' => _("Do you want to confirm deleting entries?")
 );
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/nag/config/conf.php b/horde-webmail/nag/config/conf.php
index 7809067..a0396f7 100644
--- a/horde-webmail/nag/config/conf.php
+++ b/horde-webmail/nag/config/conf.php
@@ -6,3 +6,7 @@ $conf['menu']['print'] = true;
 $conf['menu']['import_export'] = true;
 $conf['menu']['apps'] = array();
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/nag/config/menu.php b/horde-webmail/nag/config/menu.php
index 162c88c..457a10d 100644
--- a/horde-webmail/nag/config/menu.php
+++ b/horde-webmail/nag/config/menu.php
@@ -39,3 +39,7 @@
 $_menu = array();
 
 /* Add your custom entries below this line. */
+if (file_exists(dirname(__FILE__) . '/menu.local.php')) {
+  require_once(dirname(__FILE__) . '/menu.local.php');
+}
+
diff --git a/horde-webmail/nag/config/prefs.php b/horde-webmail/nag/config/prefs.php
index e14bd8a..bf2d841 100644
--- a/horde-webmail/nag/config/prefs.php
+++ b/horde-webmail/nag/config/prefs.php
@@ -238,3 +238,7 @@ $_prefs['display_tasklists'] = array(
     'shared' => false,
     'type' => 'implicit',
 );
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/turba/config/attributes.php b/horde-webmail/turba/config/attributes.php
index ce124f0..adec673 100644
--- a/horde-webmail/turba/config/attributes.php
+++ b/horde-webmail/turba/config/attributes.php
@@ -558,3 +558,7 @@ $attributes['homeLongitude'] = array(
     'type' => 'number',
     'required' => false,
 );
+if (file_exists(dirname(__FILE__) . '/attributes.local.php')) {
+  require_once(dirname(__FILE__) . '/attributes.local.php');
+}
+
diff --git a/horde-webmail/turba/config/conf.php b/horde-webmail/turba/config/conf.php
index b83f681..8f00f61 100644
--- a/horde-webmail/turba/config/conf.php
+++ b/horde-webmail/turba/config/conf.php
@@ -8,3 +8,7 @@ $conf['shares']['source'] = 'kolab';
 $conf['comments']['allow'] = true;
 $conf['documents']['type'] = 'horde';
 /* CONFIG END. DO NOT CHANGE ANYTHING IN OR BEFORE THIS LINE. */
+if (file_exists(dirname(__FILE__) . '/conf.local.php')) {
+  require_once(dirname(__FILE__) . '/conf.local.php');
+}
+
diff --git a/horde-webmail/turba/config/hooks.php b/horde-webmail/turba/config/hooks.php
index b7d0d40..d423897 100644
--- a/horde-webmail/turba/config/hooks.php
+++ b/horde-webmail/turba/config/hooks.php
@@ -38,3 +38,7 @@
 //         }
 //     }
 // }
+if (file_exists(dirname(__FILE__) . '/hooks.local.php')) {
+  require_once(dirname(__FILE__) . '/hooks.local.php');
+}
+
diff --git a/horde-webmail/turba/config/menu.php b/horde-webmail/turba/config/menu.php
index 418a4b3..49022bf 100644
--- a/horde-webmail/turba/config/menu.php
+++ b/horde-webmail/turba/config/menu.php
@@ -39,3 +39,7 @@
 $_menu = array();
 
 /* Add your custom entries below this line. */
+if (file_exists(dirname(__FILE__) . '/menu.local.php')) {
+  require_once(dirname(__FILE__) . '/menu.local.php');
+}
+
diff --git a/horde-webmail/turba/config/mime_drivers.php b/horde-webmail/turba/config/mime_drivers.php
index 729ca75..7395c13 100644
--- a/horde-webmail/turba/config/mime_drivers.php
+++ b/horde-webmail/turba/config/mime_drivers.php
@@ -19,3 +19,7 @@ $mime_drivers_map['turba']['overrides'] = array();
  * Driver specific settings. See horde/config/mime_drivers.php for
  * the format.
  */
+if (file_exists(dirname(__FILE__) . '/mime_drivers.local.php')) {
+  require_once(dirname(__FILE__) . '/mime_drivers.local.php');
+}
+
diff --git a/horde-webmail/turba/config/prefs.php b/horde-webmail/turba/config/prefs.php
index 2f7c7c3..c2d74f5 100644
--- a/horde-webmail/turba/config/prefs.php
+++ b/horde-webmail/turba/config/prefs.php
@@ -162,3 +162,7 @@ $_prefs['own_contact'] = array(
     'shared' => false,
     'type' => 'implicit'
 );
+if (file_exists(dirname(__FILE__) . '/prefs.local.php')) {
+  require_once(dirname(__FILE__) . '/prefs.local.php');
+}
+
diff --git a/horde-webmail/turba/config/sources.php b/horde-webmail/turba/config/sources.php
index 083defe..d6acb36 100644
--- a/horde-webmail/turba/config/sources.php
+++ b/horde-webmail/turba/config/sources.php
@@ -547,3 +547,7 @@ $cfgSources['favourites'] = array(
     'export' => true,
     'browse' => true,
 );
+if (file_exists(dirname(__FILE__) . '/sources.local.php')) {
+  require_once(dirname(__FILE__) . '/sources.local.php');
+}
+
-- 
tg: (b375ec9..) t/GLOBAL/HK/GW/ConfigTemplateSupport (depends on: t/GLOBAL/HK/GW/ConfigOpenPKG)
-- 
TOPGIT patch commit log
=======================

commit 1714037ef074301c0fd987b217fee5a0a1cac71b
Author: Gunnar Wrobel <p@rdus.de>
Date:   Thu Mar 26 21:08:33 2009 +0000

    Fix the inclusion statements for the motd.php files.

commit caf19136c96afa98cf1fc7f3a66cc74af6cb286a
Author: Gunnar Wrobel <p@rdus.de>
Date:   Thu Mar 26 16:44:50 2009 +0000

    Added postfix sections to all horde config files.
